File: plugin/testData/shared/viewGeneratorTest.kt
package com.lightningkite.kwift.views.shared

import android.view.View
import com.lightningkite.kwift.views.actual.ViewDependency

interface ViewGenerator {
    val title: String

    fun generate(dependency: ViewDependency): View
}


class MainViewGenerator : ViewGenerator {
    override val title: String get() = "Main"

    val stack: ObservableStack<ViewGenerator> = ObservableStack<ViewGenerator>()

    init {
        stack.push(ExampleContentViewData(stack))
    }

    override fun generate(dependency: ViewDependency): View {
        val xml = MainXml()
        val view = xml.setup(dependency)

        xml.boundViewMainContent.bindStack(dependency, stack)
        xml.boundViewTitle.bindText(stack){ it -> it.lastOrNull()?.title ?: "" }
        xml.boundViewMainBack.bindVisible(stack.transformed { it -> it.size > 1 })
        xml.boundViewMainBack.onClick(captureWeak(this) { self -> self.stack.pop(); Unit })

        return view
    }
}

File: plugin/testDataOutput2/shared/viewGeneratorTest.swift
//Package: com.lightningkite.kwift.views.shared
//Converted using Kwift2

import Foundation


public protocol ViewGenerator {
    
    var title: String { get }
    
    func generate(dependency: ViewDependency) -> View
    func generate(_ dependency: ViewDependency) -> View
}
 
 
 
public class MainViewGenerator: ViewGenerator {
    
    
    public var title: String {
        get {
            return "Main"
        }
    }
    public var stack: ObservableStack<ViewGenerator>
    
    public func generate(dependency: ViewDependency) -> View {
        var xml = MainXml()
        var view = xml.setup(dependency)
        xml.boundViewMainContent.bindStack(dependency, stack)
        xml.boundViewTitle.bindText(stack) { (it) in 
            (it.lastOrNull()?.title) ?? ""
        }
        xml.boundViewMainBack.bindVisible(stack.transformed{ (it) in 
            it.size > 1
        })
        xml.boundViewMainBack.onClick(captureWeak(self) { (self) in 
            self.stack.pop()
            ()
        })
        return view
    }
    public func generate(_ dependency: ViewDependency) -> View {
        return generate(dependency: dependency)
    }
    
    public init() {
        stack = ObservableStack<ViewGenerator>()
        stack.push(ExampleContentViewData(stack))
    }
}
 
